plugins/catalog-graph/src/components/EntityRelationsGraph/useEntityRelationNodesAndEdges.ts

```diff
@@ -57,6 +59,7 @@ export function useEntityRelationNodesAndEdges({
       maxDepth,
       kinds,
       relations,
+      func: filterFunc,
```

Comment:
I do not believe any additional filtering is necessary in `useEntityRelationNodesAndEdges.ts`. However, I'm confused below within the `useDebouce` why the `kind` is checked. Would an entity of some kind even be returned by `useEntityRelationGraph` in the first place? (I can understand why the relation types need to be checked again in this hook however)

Reply:
I think if `const entity = entities[entityRef];` evals to `undefined` as it's not in `entities` because you applied the filter there already and is ignored by `if (entity) {`, it should not be necessary to apply your filter again. But maybe a test would make sense to verify this, as I have the feeling there is a reason why `kind` and `relations` are checked again.

Reply:
Yeah let me take a closer look to be certain, thanks!